packages/salesforce-adapter/src/client/client.ts

```
@@ -1002,6 +1003,34 @@ export default class SalesforceClient implements ISalesforceClient {
     )
   }
 
+  @mapToUserFriendlyErrorMessages
+  @logDecorator()
+  @requiresLogin()
+  public async cancelMetadataValidateOrDeployTask({ taskId, taskType }: CancelServiceAsyncTaskInput): Promise<void> {
+    log.debug('Attempting to cancel %s with id %s', taskType, taskId)
+
+    const checkStatus = async (): Promise<void> => {
+      try {
+        const cancelDeployResult = await this.conn.request({
+          method: 'PATCH',
+          url: `/services/data/v${API_VERSION}/metadata/deployRequest/${taskId}`,
+          body: inspectValue({
+            deployResult: {
+              status: 'Canceling',
+            },
+          }),
+        })
+        if (_.get(cancelDeployResult, ['deployResult', 'status']) === 'Canceling') {
+          await new Promise(resolve => setTimeout(resolve, this.conn.metadata.pollInterval))
+          await checkStatus()
+        }
+      } catch (e) {
+        log.warn('Failed to cancel %s with id %s: %s', taskType, taskId, inspectValue(e))
```

I think we should return this error instead of just logging it.

---

discussed F2F. Will throw the Error here instead of returning it